mlbench_core/optim/pytorch/optim.py

```diff
@@ -499,12 +499,290 @@ def step(self, closure=None):
         return loss
 
 
+class PowerSGD(Optimizer):
+    def __init__(
+        self,
+        model=None,
+        lr=0.1,
+        weight_decay=0,
+        momentum=0,
+        dampening=0,
+        nesterov=False,
+        reuse_query=False,
+        rank=1,
```

Comment:
Shouldn't the default rank be 0 ?

Reply:
0 would transmit nothing. 1 is the correct default for low-rank compression